### PR TITLE
Refactor the Sentry plugin to use the proxy backend instead of a custom backend

### PR DESCRIPTION
**Pull request by dhenneke**
The Sentry plugin is a bit outdated and hard to use in a recent backstage instance. I changed a few things:

* Removal of the `plugin-sentry-backend` since it just proxied the request to the sentry backend. It is replaced with a `plugin-proxy-backend` configuration. I'm not sure if it was planned to add caching or aggregations etc. in the backend, but in the current state, the proxy-backend seems to be more reasonable.
* Don't differentiate the mock-backend and the production-backend via the `NODE_ENV` variable. Instead, users can replace the `sentryApiRef` with the `MockSentryApi`. tbh, I'm not sure if that still makes sense. Question would also be if we want to activate the moc-api in the example app and also have a component that has sentry configured.
* Make the `SentryIssuesWidget` usable outside of the dedicated Router-Tab-Page. Question would be if we should also remove the `Router` component completely.
* Remove the dedicated `/sentry` route. In my eyes, it doesn't really make sense since it was just configured with a hard-coded project id. Maybe in the future is would be cool to have widgets that can display issues from multiple components. But that is valid for all other plugins and should be handled in the future.
* Update the Readme.md with better usage instructions.

Fixes https://github.com/backstage/backstage/issues/3182

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)

### CODE REVIEW

File: .changeset/chatty-pens-bathe.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
+---
+'@backstage/plugin-sentry': minor
```
Comment (dhenneke) [inline .changeset/chatty-pens-bathe.md:2]: I went with `minor`. Is that correct? Do we need to announce the removal of the backend somewhere?

Comment (Fox32) [reply .changeset/chatty-pens-bathe.md:2]: I think the stability is `0` so minor is fine for this breaking change. Btw, nice migration description!

File: plugins/sentry/src/components/useProjectSlug.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Entity } from '@backstage/catalog-model';
+
+export const SENTRY_PROJECT_SLUG_ANNOTATION = 'sentry.io/project-slug';
+
+export const useProjectSlug = (entity: Entity) => {
+  return entity?.metadata.annotations?.[SENTRY_PROJECT_SLUG_ANNOTATION] ?? '';
```
Comment (Fox32) [inline plugins/sentry/src/components/useProjectSlug.ts:22]: Maybe we should return `undefined` here instead of an empty string?

**Review by Fox32 — APPROVED** [review]
Looks good. A big fresh-up for the sentry plugin!

File: .changeset/chatty-pens-bathe.md (hunk 1 above)
Comment (Rugvip) [reply .changeset/chatty-pens-bathe.md:2]: Seeing as it's being removed it doesn't matter, and this won't end up in any changelog anywhere 😅 

Perhaps it makes sense to keep the plugin package and just have a single `index.ts` that just throws an error with instructions to replace with proxy config? Then that can simply be removed in a followup PR after the release has gone out

**Review by Rugvip — COMMENTED** [review]
Thoughts on keeping a dummy `index.js` in the backend plugin that tells the user to remove it? By immediately removing the package we won't actually get a new release or changelog entry for it.

**Comment by dhenneke** [comment]
I updated it. When importing the old files, you now get an error when on starting the backend (or compiling). I also added a dummy `createRouter` export to keep `yarn tsc` happy since it won't tell the verbose error message.

<details>
<summary>yarn start-backend</summary>

```
yarn start-backend
yarn run v1.22.1
$ yarn workspace example-backend start
$ backstage-cli backend:dev
Build succeeded
webpack-internal:///../../node_modules/@backstage/plugin-sentry-backend/src/index.ts:21
throw new Error(
^

Error: The sentry-backend has been deprecated and replaced by the proxy-backend. See the changelog on how to migrate to the proxy backend: https://github.com/backstage/backstage/blob/master/plugins/sentry/CHANGELOG.md.
    at Module.eval (webpack-internal:///../../node_modules/@backstage/plugin-sentry-backend/src/index.ts:21:7)
    at eval (webpack-internal:///../../node_modules/@backstage/plugin-sentry-backend/src/index.ts:35:30)
    at Module.../../node_modules/@backstage/plugin-sentry-backend/src/index.ts (/Users/******/git/backstage/packages/backend/dist/main.js:3830:1)
    at __webpack_require__ (/Users/******/git/backstage/packages/backend/dist/main.js:749:30)
    at fn (/Users/******/git/backstage/packages/backend/dist/main.js:60:20)
    at Module.eval (webpack-internal:///./src/plugins/sentry.ts:3:90)
    at eval (webpack-internal:///./src/plugins/sentry.ts:20:30)
    at Module../src/plugins/sentry.ts (/Users/******/git/backstage/packages/backend/dist/main.js:4269:1)
    at __webpack_require__ (/Users/******/git/backstage/packages/backend/dist/main.js:749:30)
    at fn (/Users/******/git/backstage/packages/backend/dist/main.js:60:20)
    at Module.eval (webpack-internal:///./src/index.ts:11:73)
    at eval (webpack-internal:///./src/index.ts:131:30)
    at Module../src/index.ts (/Users/******/git/backstage/packages/backend/dist/main.js:4149:1)
    at __webpack_require__ (/Users/******/git/backstage/packages/backend/dist/main.js:749:30)
    at fn (/Users/******/git/backstage/packages/backend/dist/main.js:60:20)
    at Object.0 (/Users/******/git/backstage/packages/backend/dist/main.js:4293:18)
```
</details>

**Review by Rugvip — APPROVED** [review]
Thanks! 😁 

:shipit:

**Review by freben — APPROVED** [review]

**Comment by dhenneke** [comment]
I prepared a PR to remove the sentry-backend after the next release: https://github.com/backstage/backstage/pull/3621

**Pull request merged**